src/widgets/CredentialsBox.tsx

```diff
@@ -0,0 +1,56 @@
+import { Dialog } from '@jupyterlab/apputils';
+import { Widget } from '@phosphor/widgets';
+import { IGitAuth } from '../git';
+
+/**
+ * The UI for the credentials form
+ */
+export class GitCredentialsForm extends Widget
+  implements Dialog.IBodyWidget<IGitAuth> {
+  constructor(
+    textContent: string = 'Enter credentials for remote repository',
+    warningContent: string = ''
+  ) {
+    super();
+    this.node.appendChild(this.createBody(textContent, warningContent));
+  }
+
+  private createBody(textContent: string, warningContent: string): HTMLElement {
+    const node = document.createElement('div');
+    const label = document.createElement('label');
+    this._user = document.createElement('input');
```

Comment:
I am not sure how to make it work. 

HTML attribute `required` (https://www.w3schools.com/tags/att_input_required.asp) does not help here. It is possible to add this tag as an additional parameter, but nothing actually enforces it:
```
this._user.required = true;
this._password.required = true;
```

I also looked in other place: https://github.com/jupyterlab/jupyterlab/blob/9a69476fc76ab473e86edb156d7ff2a0bf88e6b5/packages/apputils/src/dialog.ts#L422 I think, it might be possible to use `accept` parameter to run a check on input fields to make sure they are not empty.

Any suggestions for implementation?

Reply:
@ktalest Thanks for the research. Unfortunately as the form tag is not directly supported for `Dialog`. There are no easy way to enforce this. I'm gonna merge this and open a separate issue for that specific point.

Note: Using the `accept` method of `Dialog` could be an option. Unfortunately the dialog is customize by injecting a widget in it and not deriving from it. So it will be quite some work.